next.config.js

```diff
@@ -21,7 +21,21 @@ const configVals = {
     basePath: basePath,
     reactStrictMode: true,
     images: {
-        domains: ['mongodb-devhub-cms.s3.us-west-1.amazonaws.com'],
+        remotePatterns: [
+            {
+                protocol: 'https',
+                hostname: '**.mongodb.com',
+            },
+            {
+                protocol: 'https',
+                hostname: 'mongodb-devhub-cms.s3.us-west-1.amazonaws.com',
+            },
+            {
+                protocol: 'https',
+                hostname: 'i.ytimg.com',
+            },
+        ],
```

Comment:
1) what is `*.mongodb.com` supposed to be for? Is it just for webimages.mongodb.com? If so why not just use that?
2) what is `i.ytimg.com`?

Might be helpful to add comments on that those are supposed to be for.

Reply:
`**.mongodb.com` covers `webimages.mongodb.com` and `www.mongodb.com` -- some images are being served from the latter. also more generally i figured that all images coming from any `mongodb.com` domain should be trusted.

`i.ytimg.com` is for youtube thumbnails. next will throw an error on most of the videos pages if this isn't here.

i'll add comments for both

Reply:
this also has me thinking, i wonder if we should add some sort of extra configurability here, maybe via an environment variable. if any different content is added through the cms (e.g. a video from a service other than youtube), we'll have to deploy a change to the codebase to allow the new domain and it'll break in the meantime. 

not sure if this is something that we expect to happen, but just a thought

Reply:
> this also has me thinking, i wonder if we should add some sort of extra configurability here, maybe via an environment variable. if any different content is added through the cms (e.g. a video from a service other than youtube), we'll have to deploy a change to the codebase to allow the new domain and it'll break in the meantime.

If we expect these to change often, then that sounds like a great idea!